controllers/gateway_controller.go

```diff
@@ -89,12 +91,21 @@ func RegisterGatewayController(
 	}
 
 	gwClassEventHandler := eventhandlers.NewEnqueueRequestsForGatewayClassEvent(mgrClient)
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&gateway_api.Gateway{}).
-		Watches(
-			&source.Kind{Type: &gateway_api.GatewayClass{}},
-			gwClassEventHandler).
-		Complete(r)
+	vpcAssociationPolicyEventHandler := eventhandlers.NewVpcAssociationPolicyEventHandler(log, mgrClient)
+	builder := ctrl.NewControllerManagedBy(mgr).
+		For(&gateway_api.Gateway{})
+	builder.Watches(&source.Kind{Type: &gateway_api.GatewayClass{}}, gwClassEventHandler)
+
+	//Watch VpcAssociationPolicy CRD if it is installed
+	if ok, err := k8s.IsGVKSupported(mgr, v1alpha1.GroupVersion.String(), v1alpha1.VpcAssociationPolicyKind); ok {
```

Comment:
nit: can you avoid nested if else statement here? Perhaps something like this:

```go
ok, err := k8s.IsGVKSupported(mgr, v1alpha1.GroupVersion.String(), v1alpha1.VpcAssociationPolicyKind)

if err != nil {
    return err
}

if ok {
    builder.Watches(&source.Kind{Type: &v1alpha1.VpcAssociationPolicy{}}, vpcAssociationPolicyEventHandler.MapToGateway())
} else {
    log.Infof("VpcAssociationPolicy CRD is not installed, skipping watch")
}
```

Reply:
thanks! changed